src/main/java/ch/njol/skript/sections/EffSecOpenInventory.java

```diff
@@ -0,0 +1,283 @@
+import org.jetbrains.annotations.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.EffectSection;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.Trigger;
+import ch.njol.skript.lang.TriggerItem;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.skript.registrations.EventValues;
+import ch.njol.skript.variables.Variables;
+import ch.njol.util.Kleenean;
+import java.util.Locale;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import ch.njol.skript.util.Version;
+
+@Name("Open/Close Inventory")
+@Description({
+	"Opens an inventory to a player. The player can then access and modify the inventory as if it was a chest that they just opened.",
+	"Note that 'show' and 'open' have different effects, 'show' will show just a view of the inventory.",
+	"Whereas 'open' will attempt to make an inventory real and usable. Like a workbench allowing recipes to work.",
+	"When using as a section. The section allows for modification of the inventory via the event-inventory."
+})
+@Examples({
+	"show crafting table to player #unmodifiable, use open instead to allow for recipes to work",
+	"open a crafting table to the player",
+	"open a loom to the player",
+	"open the player's inventory for the player",
+	"",
+	"show chest inventory to player:",
+	"\tset slot 1 of event-inventory to stone named \"example\"",
+	"\topen event-inventory to all players"
+})
+@Since("2.0, 2.1.1 (closing), 2.2-Fixes-V10 (anvil), INSERT VERSION (enchanting, cartography, grindstone, loom) & section support")
+public class EffSecOpenInventory extends EffectSection {
+
+	private static enum OpenableInventorySyntax {
+
+		ANVIL("anvil"),
+		CARTOGRAPHY("cartography [table]", Skript.methodExists(HumanEntity.class, "openCartographyTable", Location.class, boolean.class),
+				"Opening a cartography table inventory requires PaperSpigot."),
+		ENCHANTING("enchant(ment|ing) [table]", new Version(1, 14)),
+		GRINDSTONE("grindstone", Skript.methodExists(HumanEntity.class, "openGrindstone", Location.class, boolean.class),
+				"Opening a grindstone inventory requires PaperSpigot."),
+		LOOM("loom", Skript.methodExists(HumanEntity.class, "openLoom", Location.class, boolean.class),
+				"Opening a loom inventory requires PaperSpigot."),
+		SMITHING("smithing [table]", Skript.methodExists(HumanEntity.class, "openSmithingTable", Location.class, boolean.class),
+				"Opening a smithing table inventory requires PaperSpigot."),
+		STONECUTTER("stone[ ]cutter", Skript.methodExists(HumanEntity.class, "openSmithingTable", Location.class, boolean.class),
+				"Opening a stone cutter inventory requires PaperSpigot."),
+		WORKBENCH("(crafting [table]|workbench)");
```

Comment:
```suggestion
		CARTOGRAPHY("cartography [table]", Skript.methodExists(HumanEntity.class, "openCartographyTable", Location.class, boolean.class),
				"Opening a cartography table inventory requires Paper."),
		ENCHANTING("enchant(ment|ing) [table]", new Version(1, 14)),
		GRINDSTONE("grindstone", Skript.methodExists(HumanEntity.class, "openGrindstone", Location.class, boolean.class),
				"Opening a grindstone inventory requires Paper."),
		LOOM("loom", Skript.methodExists(HumanEntity.class, "openLoom", Location.class, boolean.class),
				"Opening a loom inventory requires Paper."),
		SMITHING("smithing [table]", Skript.methodExists(HumanEntity.class, "openSmithingTable", Location.class, boolean.class),
				"Opening a smithing table inventory requires Paper."),
		STONECUTTER("stone[ ]cutter", Skript.methodExists(HumanEntity.class, "openSmithingTable", Location.class, boolean.class),
				"Opening a stone cutter inventory requires Paper."),
		WORKBENCH("(crafting [table]|workbench)");
```
They stopped calling it paperspigot after 1.8 I think.